File: templates/code/python/AppConfig.py

# TODO : Add other configurations properties here
app_name = "{{ project_name }}"

{% if program_type == "spark" %}
master = "yarn"
{% if "streaming" is in feature %}streaming_time = 5{% endif %}
{% endif %}

{% if "hdfs" is in components %}
{% endif %}

{% if "none" != components %}  # HADOOP general conf{% endif %}
{% if "hdfs" is in components %}
hadooop_core_site = "/etc/hadoop/conf_cloudera_hdfs/core-site_xml"
hdfs_site_path ="/etc/hadoop/conf_cloudera_hdfs/hdfs-site_xml"{% endif %}
{% if "ozone" is in components %}ozone_site_path = "/etc/hadoop/conf_cloudera_hdfs/ozone-site_xml"{% endif %}
{% if "hbase" is in components %}hbase_site_path = "/etc/hbase/conf_cloudera_hbase/hbase-site_xml"{% endif %}

hadoop_user = "{{ hadoop_user }}"
hadoop_home = "{{ haddop_home }}"
kerberos_auth = "{{ kerberos_auth }}"
kerberos_user = "{{ kerberos_user }}"
kerberos_keytab = "{{ kerberos_keytab }}"
keystore_location = "{{ keystore_location }}"
keystore_password = "{{ keystore_password }}"
keystore_keypassword = "{{ keystore_key_password }}"
truststore_location = "{{ truststore_location }}"
truststore_password = "{{ truststore_password }}"
{% endif %}

{% if "hdfs" is in components %}
hdfs_nameservice = "{{ hdfs_nameservice }}"
hdfs_port =  "8020"
hdfs_home_dir = "{{ hdfs_workdir }}"
{% endif %}

{% if "hbase" is in components %}  # HBASE
hbase_zookeeper_quorum = "{{ zookeeper_quorum }}"
hbase_zookeeper_port = "2181"
hbase_zookeeper_znode = "/hbase"
{% endif %}

{% if "ozone" is in components %}  # OZONE
ozone_nameservice = "{{ ozone_nameservice }}"
{% endif %}

{% if "hive" is in components %}  # HIVE
hive_zookeeper_server = "{{ zookeeper_quorum }}"
hive_zookeeper_port = "2181"
hive_zookeeper_znode = "hiveserver2"
{% endif %}

{% if "solr" is in components %}  # SOLR
solr_server = "{{ solr_server }}"
solr_port = "8983"
{% endif %}

{% if "kafka" is in components %}  # KAFKA
kafka_brokers = "{{ kafka_broker }}"
security_protocol = "{{ kafka_security_protocol }}"
schema_registry_url = "{{ schema_registry }}"
{% endif %}

{% if "kudu" is in components %}  # KUDU
kudu_master = "{{ kudu_master }}"
{% endif %}

File: templates/code/scala/AppConfig.scala
/**
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package {{ package_name }}.config

import com.typesafe.config.Config
import com.typesafe.config.ConfigFactory


object AppConfig {

  val conf: Config = ConfigFactory.load()

  val name = conf.getString("appName")

  {% if program_type == "spark" %}
  val master = conf.getString("master")
  {% if feature is not none and "streaming" is in feature %}val streamingTime = conf.getInt("streamingTime"){% endif %}
  {% endif %}

  {% if "none" != components %}
  {% if "hdfs" is in components %}val coreSitePath = conf.getString("hadoop.core.site.path")
  val hdfsSitePath = conf.getString("hadoop.hdfs.site.path"){% endif %}
  {% if "ozone" is in components %}val ozoneSitePath = conf.getString("hadoop.ozone.site.path"){% endif %}
  {% if "hbase" is in components %}val hbaseSitePath = conf.getString("hadoop.hbase.site.path"){% endif %}

  val hadoopUser = conf.getString("hadoop.user")
  val hadoopHome = conf.getString("hadoop.home")
  val kerberosAuth = conf.getString("kerberos.auth")
  val kerberosUser = conf.getString("kerberos.user")
  val kerberosKeytab = conf.getString("kerberos.keytab")
  val keystoreLocation = conf.getString("keystore.location")
  val keystorePassword = conf.getString("keystore.password")
  val keystoreKeyPassword = conf.getString("keystore.keypassword")
  val truststoreLocation = conf.getString("truststore.location")
  val truststorePassword = conf.getString("truststore.password")
  {% endif %}

  {% if "hdfs" is in components %}
  val hdfsNameservice = conf.getString("hdfs.nameservice")
  val port = conf.getString("hdfs.port")
  val hdfsHomeDir = conf.getString("hdfs.home_dir")
  {% endif %}

  {% if "hbase" is in components %}
  val hbaseZookeeperQuorum = conf.getString("hbase.zookeeper.quorum")
  val hbaseZookeeperPort = conf.getString("hbase.zookeeper.port")
  val habseZookeeperZnode = conf.getString("hbase.zookeeper.znode")
  {% endif %}

  {% if "ozone" is in components %}
  val ozoneNameService = conf.getString("ozone.nameservice")
  {% endif %}

  {% if "hive" is in components %}
  val hiveZookeeperQuorum = conf.getString("hive.zookeeper.quorum")
  val hiveZookeeperPort = conf.getString("hive.zookeeper.port")
  val hiveZookeeperZnode = conf.getString("hve.zookeeper.znode")
  {% endif %}

  {% if "solr" is in components %}
  val solrServer = conf.getString("solr.server")
  val solrPort = conf.getString("solr.port")
  {% endif %}

  {% if "kafka" is in components %}
  val kafkaBrokers = conf.getString("kafka.brokers")
  val kafkaSecurityProtocol = conf.getString("kafka.security_protocol")
  val schemaRegistryUrl = conf.getString("kafka.schema_registry.url")
  {% endif %}

  {% if "kudu" is in components %}
  val kuduMaster = conf.getString("kudu.master")
  {% endif %}

}
